pybamm/experiments/experiment.py

```diff
@@ -143,37 +162,112 @@ def read_string(self, cond):
         else:
             period = self.period
         # Read instructions
-        if "for" in cond and "or until" in cond:
-            # e.g. for 3 hours or until 4.2 V
-            cond_list = cond.split()
-            idx_for = cond_list.index("for")
-            idx_until = cond_list.index("or")
-            electric = self.convert_electric(cond_list[:idx_for])
-            time = self.convert_time_to_seconds(cond_list[idx_for + 1 : idx_until])
-            events = self.convert_electric(cond_list[idx_until + 2 :])
-        elif "for" in cond:
-            # e.g. for 3 hours
-            cond_list = cond.split()
-            idx = cond_list.index("for")
-            electric = self.convert_electric(cond_list[:idx])
-            time = self.convert_time_to_seconds(cond_list[idx + 1 :])
-            events = None
-        elif "until" in cond:
-            # e.g. until 4.2 V
+        if "Run" in cond:
             cond_list = cond.split()
-            idx = cond_list.index("until")
-            electric = self.convert_electric(cond_list[:idx])
-            time = None
-            events = self.convert_electric(cond_list[idx + 1 :])
-        else:
-            raise ValueError(
-                """Operating conditions must contain keyword 'for' or 'until'.
-                For example: {}""".format(
-                    examples
+            if "at" in cond:
+                raise ValueError(
+                    """Instruction must be
+                    For example: {}""".format(
+                        examples
+                    )
+                )
+                # Check for Events
+            elif "for" in cond and "or until" in cond:
+                # e.g. for 3 hours or until 4.2 V
+                idx_for = cond_list.index("for")
+                idx_until = cond_list.index("or")
+                end_time = self.convert_time_to_seconds(
+                    cond_list[idx_for + 1:idx_until])
+                ext_drive_cycle = self.extend_drive_cycle(drive_cycles[cond_list[1]],
+                                                          end_time)
+                electric = (ext_drive_cycle[:, 1], "Drive")
+                time = self._np.append(1, self._np.diff(ext_drive_cycle[:, 0]))
+                period = time
+                events = self.convert_electric(cond_list[idx_until + 2 :])
+            elif "for" in cond:
+                # e.g. for 3 hours
+                idx = cond_list.index("for")
+                end_time = self.convert_time_to_seconds(cond_list[idx + 1 :])
+                ext_drive_cycle = self.extend_drive_cycle(drive_cycles[cond_list[1]],
+                                                          end_time)
+                electric = (ext_drive_cycle[:, 1], "Drive")
+                time = self._np.append(1, self._np.diff(ext_drive_cycle[:, 0]))
+                period = time
+                events = None
+            elif "until" in cond:
```

Comment:
Dealing with events in drive cycles is challenging numerically. When running real experiments people usually choose a drive cycle that is short enough that the voltage limits won't be triggered (e.g. only 20% discharge). So I don't think we need the "until" condition for drive cycles.

Reply:
The implementation of the 'until' condition in the simulation class was the only limitation in commit (ffd340c) of this PR. Nevertheless, I will remove the implementation of until event for drive cycles from the experiment class in the next commit